src/interfaces/pyo3/macros.rs

```diff
@@ -0,0 +1,33 @@
+/// Implements the basic functionalities of a key in python.
+///
+/// - implements `deep_copy`
+/// - converts to and from `PyBytes`
+///
+/// # Parameters
+///
+/// - `type_name`   : name of the key type
+macro_rules! impl_key_byte {
+    ($py_type:ty, $rust_type:ty) => {
+        #[pymethods]
+        impl $py_type {
+            /// Clones the key
+            pub fn deep_copy(&self) -> Self {
+                Self(self.0.clone())
+            }
+
+            /// Converts key to bytes
+            pub fn to_bytes(&self, py: Python) -> PyResult<PyObject> {
+                Ok(convert_to_pybytes(&self.0.try_to_bytes()?, py))
+            }
+
+            /// Reads key from bytes
+            #[classmethod]
+            pub fn from_bytes(_cls: &PyType, key_bytes: Vec<u8>) -> PyResult<Self> {
```

Comment:
You don't need ownership here.
```suggestion
            pub fn from_bytes(_cls: &PyType, key_bytes: &[u8]) -> PyResult<Self> {
```